libkineto/src/CudaDeviceProperties.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "CudaDeviceProperties.h"
+
+#include <vector>
+
+#include <cuda_runtime.h>
+#include <cuda_occupancy.h>
+
+namespace KINETO_NAMESPACE {
+
+std::vector<cudaOccDeviceProp> occProps;
+
+void initOccDeviceProp() {
```

Comment:
How about calling this lazily from within this file so it doesn't have to be explicitly initialized:

namespace {

std::vector<cudaOccDeviceProp> createOccDeviceProps() {
  std::vector<cudaOccDeviceProp> res;
  ...
  return res;
}

std::vector<cudaOccDeviceProps>& occDeviceProps() {
  static std::vector<cudaOccDeviceProp> props = createOccDeviceProps();
  return props;
}

} // namespace

Then in getKernelOccupancy replace occProps with occDeviceProps()

Reply:
Thanks! Fixed.